shell/common/shell.cc

```diff
@@ -1229,4 +1237,15 @@ Rasterizer::Screenshot Shell::Screenshot(
   return screenshot;
 }
 
+bool Shell::WaitForFrameRender(fml::TimeDelta timeout) {
```

Comment:
Another (probably simpler) approach to consider:

1. Create a `fml::AutoResetWaitableEvent latch` here
2. Push a task to the GPU thread that adds this latch to a vector owned by Shell. That vector is only supposed to be accessed in the GPU thread.
3. In `Shell::OnFrameRasterized` (which runs on the GPU thread), notify all latches in the vector and clear that vector.
3. Wait for the latch (with optional timeout).

I think it will save us from reasoning through the lock and condition variables, and just rely on our existing task runner mechanism.